src/core/coap/coap.cpp

```diff
@@ -259,42 +259,34 @@ void CoapBase::HandleRetransmissionTimer(Timer &aTimer)
 
 void CoapBase::HandleRetransmissionTimer(void)
```

Comment:
The diff here looks bigger than the actual change. Here are some points on what is changed:
- The `while()` iteration loop is changed to a ` for()`: this allows us to use `continue` to skip over entries.
- We move up the check for entries that are expired and use `continue` to skip over next entry in loop if no need to retx the entry.
- In the case that retx is required for an expired entry, the `mNextTimerShot` gets updated.
- The code which updates the next timer fire time (`nextTime` variable) is now moved at the end and is common for both cases (where entry is not expired or where the entry is expired and we need a retx).
- After the loop, we check if the we need to start the timer with ` nextTime`.

A similar pattern is used in other modules.